### PR TITLE
Refactor to improve types for `selector-(class|combinator|descendant|disallowed|id)-*` rules

### PR DESCRIPTION
**Pull request by ybiquitous**
> Which issue, if any, is this issue related to?

Part of #4496

> Is there anything in the PR that needs further explanation?

- Remove `// @ts-nocheck` to enable type-checking.
- Improve the code a bit for type safety.

### CODE REVIEW

File: lib/rules/selector-descendant-combinator-no-non-space/index.js [hunk 1]
```diff
@@ -52,6 +51,7 @@ function rule(expectation, options, context) {
 					) {
 						if (context.fix && /^\s+$/.test(value)) {
 							hasFixed = true;
+							// @ts-expect-error -- TS2339: Property 'raws' does not exist on type 'Combinator'.
 							combinatorNode.raws.value = ' ';
```
Comment (ybiquitous) [inline lib/rules/selector-descendant-combinator-no-non-space/index.js:55]: [note] This error is probably caused by the insufficient type definition of `postcss-selector-parser`:

```
lib/rules/selector-descendant-combinator-no-non-space/index.js:55:23 - error TS2339: Property 'raws' does not exist on type 'Combinator'.
```

Comment (ybiquitous) [reply lib/rules/selector-descendant-combinator-no-non-space/index.js:55]: I've opened https://github.com/postcss/postcss-selector-parser/issues/253

**Review by jeddy3 — APPROVED** [review]
Good stuff. LGTM, thanks!

File: lib/rules/selector-descendant-combinator-no-non-space/index.js (hunk 1 above)
Comment (ybiquitous) [reply lib/rules/selector-descendant-combinator-no-non-space/index.js:57]: I've bump up `postcss-selector-parser` and fixed the type error via b95b60db.
The new version includes only my patch. 😃 

- https://github.com/postcss/postcss-selector-parser/releases/tag/v6.0.9
- https://github.com/postcss/postcss-selector-parser/compare/v6.0.8...v6.0.9

(thanks to @alexander-akait 👍🏼 )

Comment (ybiquitous) [reply lib/rules/selector-descendant-combinator-no-non-space/index.js:57]: @jeddy3 Sorry, could you please re-review the change when you have time?

**Review by jeddy3 — APPROVED** [review]
Good work on the upstream patch.

Everything here LGTM.

**Comment by ybiquitous** [comment]
@jeddy3 Thank you so much! 😄

**Pull request merged**